File: Linked List/PalindromeLL.java
// Leetcode 234

package Linked List;

public class PalindromeLL {
    public boolean isPalindrome(ListNode head) {
        ListNode left = head;
        List<Integer> numList = new ArrayList<>();

        while(left != null){
            numList.add(left.val);
            left = left.next;
        }
        
        int l = 0, r = numList.size() - 1;
        while (l < r){
            if (numList.get(l) != numList.get(r)) return false;
            l++;
            r--;
        }
        return true;
    }

    public static void main(String[] args) {
        
    }
}

File: LinkedList/Palindrome LL.py
# Leetcode 234

class Solution(object):
    def isPalindrome(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: bool
        """
        left = head
        numList = []

        while (left != None):
            numList.append(left.val)
            left = left.next
        
        l = 0
        r = len(numList) - 1
        while (l < r):
            if (numList[l] != numList[r]): return False
            l += 1
            r -= 1
        
        return True